### ISSUE TITLE
[FEA] Nulls pushdowns for LIST columns

### ISSUE THREAD

**Issue by devavret** [rapidsai/cudf#10291]
There are places in libcudf that expect a LIST column that has nulls in parent properly applied to the child.[1] i.e. if a list is null then it should also be empty and thus the offsets describing it should be same.

Similar to the `superimpose_parent_nulls` function for STRUCT columns, we need a function that can pushdown the nulls in a list column and compact a child when it had non-empty nulls.

This is also a blocker for #10289 

[1] Parquet writer depends on the nulls properly applied to leaf child. When writing leaf child columns of lists, the range to write in a particular page is determined using the offsets. e.g. if row 0 to 5 are to be written to page 1, then the selected children are offsets[0] to offset[5]. If row 2 is null but offset[2] < offset[2+1] then those extraneous children also end up in the page.

**Comment by jrhemstad** [comment]
For posterity, can you summarize how a non-empty null list breaks things? 

I liked your naming of "sanitize" as opposed to "superimpose" or "pushdown" in this scenario. 

So something like `sanitize_null_lists`. 

Or maybe "normalize"? `normalize_null_lists`?

**Comment by jrhemstad** [comment]
I think the easiest way to implement this is to simply perform an identity gather (i.e., gather map `[0,n)`).

Unfortunately, I spoke with @nvdbaranec  and `gather` won't filter out non-empty lists currently. But it should be easier to modify `gather` than to create a whole new function just for stripping out non-empty null lists.

In addition, I'd like to see two additional APIs for determining if sanitization is necessary:
1. An approximate/fast one that only checks the null counts of all the nested list columns
2. An exact/slower one that launches a kernel to inspect the offsets/null masks to determine if non-empty null lists exist.

**Comment by devavret** [comment]
> if a file contains a dirty list column, we would sanitize it as part of reading it and returning it

This is always true for parquet, because parquet has a different representation which we convert from and i don't think we can create a dirty list from it, ever. @nvdbaranec, right?

**Comment by vuule** [comment]
I'm pretty sure ORC by definition cannot have elements in a null list. Content of null elements is never encoded in the file (which is 
 a pain for other reasons). @rgsl888prabhu, can you confirm?

**Comment by sameerz** [comment]
Still needed.
